R/IN22.R

```diff
@@ -0,0 +1,139 @@
+  data,
+  "variable_used",
+  "age_used",
+  "weight_used",
+  "year_used",
+  "label"
+)
+
+
+##########################################
+# SECTION: By socio-economic background
+
+section_chart_type = "seb"
+section_csv_name = "SEB"
+
+data_for_section = get_data_for_chart_type(data, section_chart_type)
+
+#################
+# CHART FORMAT
+
+data_for_section = data_frame__sort_rows_with_specific_values(
+  data_frame = data_for_section,
+  column_1 = "secondary_split_value",
+  values_1 = occupational_class_order,
+  # column_2 = "secondary_split_value",
+  # values_2 = occupational_class_order
+) %>% filter(!is.na(secondary_split_value))
+
+
+csv_filename = generate_csv_file_name(split = section_csv_name, format = "chart")
+save_data_frame(data_for_section, csv_filename)
+
+#################
+# TABLE FORMAT
+
+
+pivot_longer_w_ci = data_for_section %>% rename(
+  c("Percent" = "value", "Lower CI" = "lci", "Upper CI" = "uci")) %>% 
+  pivot_longer(cols=c("Percent", "Lower CI", "Upper CI"))
+
+pivot_table = pivot_table__create(
+  pivot_table_source = pivot_longer_w_ci,
+  pivot_columns_column_name = "name",
+  pivot_rows_column_name = "secondary_split_value",
+  pivot_cells_column_name = "value",
+  pivot_table_name = "Socio-economic background",
+  pivot_table_rows_order_values = rev(occupational_class_order),
+  pivot_table_columns_order_values = c("Percent", "Lower CI", "Upper CI"),
+  pivot_table_column_names_suffix = " (%)"
+) 
+
+csv_filename = generate_csv_file_name(split = section_csv_name, format = "table")
+save_data_frame(pivot_table, csv_filename)
+
+
+##########################################
+# SECTION: Time series
+
+section_chart_type = "trend_moving_average"
+section_csv_name = "moving-average"
+
+
+data_for_section = get_data_for_chart_type(data, section_chart_type)
+
+data_for_section$secondary_split_value <- replace_na(data_for_section$secondary_split_value, "Total") 
+
+#################
+# CHART FORMAT
+
+data_for_section = data_frame__sort_rows_with_specific_values(
+  data_frame = data_for_section,
+  column_1 = "primary_split_value",
+  values_1 = NULL,
+  column_2 = "secondary_split_value",
+  values_2 = occupational_class_order
+) %>% 
+  # mutate_at("primary_split_value", as.numeric) %>% 
+  mutate(
+    primary_split_value = as.numeric(primary_split_value),
+    average_window = paste(primary_split_value - 2, primary_split_value, sep=" to "),
+    )
+
+csv_filename = generate_csv_file_name(split = section_csv_name, format = "chart")
+save_data_frame(data_for_section, csv_filename)
+
+#################
+# TABLE FORMAT
+
+pivot_table = pivot_table__create(
+  pivot_table_source = data_for_section,
+  pivot_columns_column_name = "secondary_split_value",
+  pivot_rows_column_name = "average_window",
+  pivot_cells_column_name = "value",
+  pivot_table_name = "Year",
+  # pivot_table_rows_order_values = rev(occupational_class_order),
+  # pivot_table_columns_order_values = rev(occupational_class_order),
```

Comment:
For the Moving Average data table:
* we normally sort dates from most-recent to least-recent (in some cases we have so many rows that some are initially truncated, so it's nice to see the newest data first)
* We'd like to sort the columns by occupational class order.
Maybe we need a new ordering for this which includes "Total" or maybe a way of adding a custom final value (e.g. maybe it's sometimes "Total" and sometimes "All ethnicities")

Reply:
I've added in these sortings, let me know what you think

Reply:
Looks great! Thanks 😄